p2p/protocols/protocol.go

```diff
@@ -157,6 +146,10 @@ type Spec struct {
 	initOnce sync.Once
 	codes    map[reflect.Type]uint64
 	types    map[uint64]reflect.Type
+
+	// if the protocol allows for extending the p2p msg to propagate context
+	// even if set to true context will propagate only if the remote peer supports it
```

Comment:
The comment looks to be right if the property would be called `EnableContext` . I would suggest to adjust the comment or rename `DisableContext` to `EnableContext`